### DISCUSSION TITLE
pls help me.. arudino diagram..

### DISCUSSION THREAD

**Discussion by rootyou07** [community/community#141727] [Programming Help]
### Body

pls help...
Please tell me the Arduino circuit diagram that can operate the zp12s4010h part according to humidity.

srd-05vdc-sl-c(relay module), dht sensor, arduino, zp12s4010h(pan)

### Guidelines

- [X] I have read and understood this category's [guidelines](https://github.com/orgs/community/discussions/51384) before making this post.

**Comment by Danidukiyu** [comment]
## **Circuit Connections:**
DHT Sensor Connections:
VCC: Connect to Arduino 5V
GND: Connect to Arduino GND
DATA: Connect to a digital pin on Arduino (e.g., pin 2)

## **Relay Module Connections:**
VCC: Connect to Arduino 5V
GND: Connect to Arduino GND
IN: Connect to another digital pin on Arduino (e.g., pin 7)

## **ZP12S4010H Pan Connections:**
Connect the NO (Normally Open) terminal of the relay to one terminal of the pan.
Connect the other terminal of the pan to the power supply.
Connect the COM (Common) terminal of the relay to the power supply (matching the voltage of the pan).

Do you want me to provide a visual Diagram too? (this should give you enough base to Start and I can write a basic code for operating DHT sensor too)

**Reply by Danidukiyu** [reply]
#include <DHT.h>

#define DHTPIN 2     // DHT sensor data pin
#define DHTTYPE DHT11   // Change to DHT22 if you're using a DHT22
#define RELAYPIN 7   // Relay control pin

DHT dht(DHTPIN, DHTTYPE);

void setup() {
  Serial.begin(9600);
  dht.begin();
  pinMode(RELAYPIN, OUTPUT);
}

void loop() {
  // Wait a few seconds between measurements.
  delay(2000);

  // Read humidity
  float h = dht.readHumidity();
  
  // Check if any reads failed and exit early (to try again).
  if (isnan(h)) {
    Serial.println("Failed to read from DHT sensor!");
    return;
  }

  Serial.print("Humidity: ");
  Serial.print(h);
  Serial.println("%");

  // Control relay based on humidity threshold
  if (h > 60) { // Set your desired humidity threshold here
    digitalWrite(RELAYPIN, HIGH); // Turn on the relay (and the pan)
    Serial.println("Relay ON");
  } else {
    digitalWrite(RELAYPIN, LOW); // Turn off the relay
    Serial.println("Relay OFF");
  }
}

**Comment by tpmst** [comment]
Hello @rootyou07,

what exactly do you need?

**Comment by NabilMch** [comment]
Hey! Here’s a simple setup to get that fan working based on humidity:

1. **DHT Sensor**: Hook the data pin to a digital pin on your Arduino (like **D2**). This will give you the humidity readings.
   
2. **Relay Module (SRD-05VDC-SL-C)**: Connect the relay control pin to another digital pin (say, **D3**). This relay will switch your fan on and off.
   
3. **Fan (ZP12S4010H)**: Wire the fan’s power through the relay (connect it to the **NO** terminal), so when the relay activates, the fan kicks on.

For the code, just set it up to read the humidity and turn the fan on if it goes above a certain level. When it drops, the fan switches off. Easy setup!

Let me know if you need help with the code part—happy to help!
